addons/website/static/src/builder/plugins/options/carousel_slides_option_plugin.js

```diff
@@ -0,0 +1,85 @@
+import { BuilderAction } from "@html_builder/core/builder_action";
+import { SNIPPET_SPECIFIC_END } from "@html_builder/utils/option_sequence";
+import { Plugin } from "@html_editor/plugin";
+import { withSequence } from "@html_editor/utils/resource";
+import { registry } from "@web/core/registry";
+
+export class CarouselSlidesOptionPlugin extends Plugin {
+    static id = "carouselSlidesOption";
+    resources = {
+        builder_options: [
+            withSequence(SNIPPET_SPECIFIC_END, {
+                template: "website.CarouselSlidesOption",
+                selector: ".carousel .carousel-item",
+                exclude: ".s_image_gallery .carousel-item",
+            }),
+        ],
+        builder_actions: {
+            MakeSlideClickableAction,
+            SetSlideAnchorUrlAction,
+        },
+        clean_for_save_handlers: this.cleanForSave.bind(this),
+    };
+
+    /**
+     * Handler called before saving changes in the website editor.
+     *
+     * @param {HTMLElement} root
+     */
+    cleanForSave({ root }) {
+        // Remove `clickable-slide` class from slides when there is no link
+        // element.
+        const noLinkSlideEls = root.querySelectorAll(
+            ".carousel-item.clickable-slide:not(:has(.slide-link))"
+        );
+        for (const slideEl of noLinkSlideEls) {
+            slideEl.classList.remove("clickable-slide");
+        }
+    }
+}
+
+class MakeSlideClickableAction extends BuilderAction {
+    static id = "makeSlideClickable";
+    setup() {
+        this.preview = false;
+    }
+    clean({ editingElement }) {
+        // Remove unnecessary link from the slide when toggled off.
+        const linkEl = editingElement.querySelector("a.slide-link");
+        linkEl?.remove();
+    }
+}
+
+class SetSlideAnchorUrlAction extends BuilderAction {
+    static id = "setSlideAnchorUrl";
+    setup() {
+        this.preview = false;
+    }
+    apply({ editingElement, value }) {
+        const url = value;
+        const linkEl = editingElement.querySelector("a.slide-link");
+
+        // Remove link if no URL
+        if (!url) {
+            linkEl.remove();
+            return;
+        }
+        // If link already exists, just update its href
+        if (linkEl) {
+            linkEl.setAttribute("href", url);
+            return;
+        }
+        // Otherwise, create and insert the link
+        const anchorEl = document.createElement("a");
```

Comment:
This PR lacks meaningful comments at the class-level. Those comments here are however useless: they describe obvious two-lines of code.